### PR TITLE
(DRON-392) cascade deletes on purge

### PR DESCRIPTION
**Pull request by tphoney**
## Commit Checklist

Thank you for creating a pull request! To help us review / merge this can you make sure that your PR adheres as much as possible to the following.

### The Basics

- Commit is a single logical unit of work, only use multiple commits if doing different tasks
- Commit does not include commented out code or unneeded files
- rebase of main branch

### The Content

- Must include testing for bug or feature
- Must include appropriate documentation changes if it is introducing a new feature or changing existing functionality
- Must pass existing test suites

### The Commit Message

- Short meaningful description (ex: remove deprecated steps)
- Uses the imperative, present tense: "change", not "changed" or "changes"
- Includes motivation for the change, and contrasts its implementation with the previous behavior

### The Pull Request

- What is the reason for this change
- Example usage of the failure for a bug, or configuration and expected output for a feature
- Steps to test the change

### CODE REVIEW

File: core/stage.go [hunk 1]
```diff
@@ -77,6 +77,9 @@ type (
 
 		// Update persists an updated stage to the datastore.
 		Update(context.Context, *Stage) error
+
+		// Purge removes orpaned stages from the datastore.
+		Purge(context.Context) error
```
Comment (bradrydzewski) [inline core/stage.go:82]: Instead of having an explicit Stage purge function, I would handle all the cascading purges in the [Build.Purge](https://github.com/harness/drone/blob/master/store/build/build.go#L414) function.  This way, if we ever decide to switch to `ON DELETE` constraints, all code changes are isolated to the database code.

**Review by bradrydzewski — APPROVED** [review]

**Review by bradrydzewski — CHANGES_REQUESTED** [review]
Just a minor suggestion to encapsulate cascading purges in the Build purge function.

File: store/build/build_test.go [hunk 2]
```diff
@@ -25,7 +26,7 @@ func TestBuild(t *testing.T) {
 	}
 	defer func() {
 		dbtest.Reset(conn)
-		dbtest.Disconnect(conn)
+		_ = dbtest.Disconnect(conn)
```
Comment (eoinmcafee00) [inline store/build/build_test.go:29]: why was this change required?

**Review by eoinmcafee00 — APPROVED** [review]

**Comment by kosssi** [comment]
Why you don't respect your own rules? I thinks it's very bad pull request to not explain the reason of change.
